block/executor.go

```diff
@@ -194,45 +181,8 @@ func (e *Executor) commit(ctx context.Context, state *types.State, block *types.
 	return resp.Data, resp.RetainHeight, err
 }
 
-func (e *Executor) validateBlock(state types.State, block *types.Block) error {
-	err := block.ValidateBasic()
-	if err != nil {
-		return err
-	}
-	if block.Header.Version.App != state.Version.Consensus.App ||
-		block.Header.Version.Block != state.Version.Consensus.Block {
-		return errors.New("block version mismatch")
-	}
-	if state.LastBlockHeight <= 0 && block.Header.Height != uint64(state.InitialHeight) {
-		return errors.New("initial block height mismatch")
-	}
-	if state.LastBlockHeight > 0 && block.Header.Height != uint64(state.LastStoreHeight)+1 {
-		return errors.New("block height mismatch")
-	}
-	if !bytes.Equal(block.Header.AppHash[:], state.AppHash[:]) {
-		return errors.New("AppHash mismatch")
-	}
-	if !bytes.Equal(block.Header.LastResultsHash[:], state.LastResultsHash[:]) {
-		return errors.New("LastResultsHash mismatch")
-	}
-
-	return nil
-}
-
-func (e *Executor) validateCommit(proposer *types.Sequencer, commit *types.Commit, header *types.Header) error {
-	abciHeaderPb := abciconv.ToABCIHeaderPB(header)
-	abciHeaderBytes, err := abciHeaderPb.Marshal()
-	if err != nil {
-		return err
-	}
-	if err = commit.Validate(proposer, abciHeaderBytes); err != nil {
-		return err
-	}
-	return nil
-}
-
-// Execute executes the block and returns the ABCIResponses.
-func (e *Executor) Execute(ctx context.Context, state types.State, block *types.Block) (*tmstate.ABCIResponses, error) {
+// ExecuteValidBlock executes the block and returns the ABCIResponses.
+func (e *Executor) ExecuteValidBlock(ctx context.Context, state types.State, block *types.Block) (*tmstate.ABCIResponses, error) {
```

Comment:
I think the `Valid` in the name is bit redundant.
the execution will fail and return err if execution will fail due to invalid block

Reply:
fixed